lib/cron/synth-switch.ts

```diff
@@ -41,14 +79,170 @@ const handler: ScheduledHandler = async (_event: EventBridgeEvent<string, void>)
+        } else {
+          hasPriceImprovement = BigNumber.from(order.classic_amountingasadjusted).gt(order.settledAmountIn);
+        }
+
+        if (!(key in tradeOutcomesByKey)) {
+          tradeOutcomesByKey[key] = {
+            pos: 0,
+            neg: 0,
+          };
+        }
+        if (hasPriceImprovement) {
+          tradeOutcomesByKey[key].pos++;
+        } else {
+          tradeOutcomesByKey[key].neg++;
+        }
+      }
 
+      Object.keys(tradeOutcomesByKey).forEach(async (key) => {
+        const { pos, neg } = tradeOutcomesByKey[key];
+        const totalOrders = pos + neg;
+        if (totalOrders >= MINIMUM_ORDERS) {
+          if (neg / totalOrders >= DISABLE_THRESHOLD) {
+            // TODO: update tradeSizes with new TokenConfig schema
+            log.info(
+              {
+                key,
+                ordersWithNegativePriceImprovement: neg,
+                ordersWithPositivePriceImprovement: pos,
+                totalOrders,
+              },
+              'Disabling synthethics for trade'
+            );
+            await synthSwitchEntity.putSynthSwitch(SwitchRepository.parseKey(key), config.tradeSizes[0], false);
+            return;
+          }
+        }
+        if (pos > 0) {
+          log.info(
+            {
+              key,
+              ordersWithNegativePriceImprovement: neg,
+              ordersWithPositivePriceImprovement: pos,
+              totalOrders,
+            },
+            'Enabling synthethics for trade'
+          );
+          // TODO: update tradeSizes with new TokenConfig schema
+          await synthSwitchEntity.putSynthSwitch(SwitchRepository.parseKey(key), config.tradeSizes[0], true);
+        }
+      });
+    }
+  }
+
+  // create view
   try {
     const createViewResponse = await client.send(
-      new ExecuteStatementCommand({ ...sharedConfig, Sql: fillMetricsSql(configs) })
+      new ExecuteStatementCommand({ ...sharedConfig, Sql: CREATE_COMBINED_URA_RESPONSES_VIEW_SQL })
     );
     stmtId = createViewResponse.Id;
+  } catch (e) {
+    log.error({ error: e }, 'Failed to send create view command');
+    throw e;
+  }
+  for (;;) {
+    const status = await client.send(new DescribeStatementCommand({ Id: stmtId }));
+    if (status.Error) {
+      log.error({ error: status.Error }, 'Failed to create view');
+      throw new Error(status.Error);
+    }
+    if (status.Status === StatusString.ABORTED || status.Status === StatusString.FAILED) {
+      log.error({ error: status.Error }, 'Failed to execute create view command');
+      throw new Error(status.Error);
+    } else if (
+      status.Status === StatusString.PICKED ||
+      status.Status === StatusString.STARTED ||
+      status.Status === StatusString.SUBMITTED
+    ) {
+      await sleep(2000);
+    } else if (status.Status === StatusString.FINISHED) {
+      break;
+    } else {
+      log.error({ error: status.Error }, 'Unknown status');
+      throw new Error(status.Error);
+    }
+  }
```

Comment:
nit: looks like this for loop code for waiting on the SQL command to return is repeated down below. Anyway we could turn this into some kind of function for reuse? no worries if that gets too tricky. I know there's some logic we are doing inside of it to so maybe not the cleanest thing to consolidate.

Reply:
hm yeah could do something like `handleStmtResult`, down for a fast follow on this? like you said I think we have some logic in the FINISHED state for one use case